routers/user.js

```diff
@@ -35,30 +35,78 @@ class User {
                 lastName: context.userInfo.lastName,
                 createdAt: session_currentTime,
                 updateAt: session_currentTime
-                }
+            }
             await this.userCollection.insert(user);
+        } else {
+            user = result[0];
         }
-        else{
-            user =  result[0];
-        }
-        if (result.matchedCount < 1){
+        if (result.matchedCount < 1) {
             let error = "there is an error getting the result";
             console.error(error)
             throw new Error(error)
-          } 
+        }
         context.userInfo = {
             ...user,
             ...context.userInfo
         }
         return user
     }
-    
 
-    async updateMyUser(parm, context) {
+    async updateMyUser(params, context) {
+        let session_currentTime = getCurrentTimeYYYYMMDDSS();
+        let user
+
+        params.userInfo._id = context.userInfo._id;
+        params.userInfo.updateAt = context.userInfo.updateAt;
+        const target_obj ={
+            _id: context.userInfo._id,
+            firstName: params.userInfo.firstName,
+            lastName: params.userInfo.lastName,
+            updateAt: session_currentTime
+        }
+        const original_result = await this.userCollection.find(context.userInfo._id);
+
+        const original_obj = {
+            _id: original_result[0]._id,
+            firstName: original_result[0].firstName,
+            lastName: original_result[0].lastName,
+            updateAt: session_currentTime
+
+        } 
+
+        let current_obj = original_obj
+        let update_result 
+
+        if ((target_obj.firstName != original_obj.firstName)||(target_obj.lastName != original_obj.lastName)){
```

Comment:
The check to see if firstName and lastName is not needed because we will always update the "updateAt" field when a request is received even if the data is unchanged